src/components/ContentUploader/ContentUploader.js

```diff
@@ -426,12 +587,23 @@ class ContentUploader extends Component<DefaultProps, Props, State> {
         items[items.indexOf(item)] = item;
 
         // Broadcast that there was an error uploading a file
-        onError({
-            file,
-            error
-        });
+        if (windowView) {
+            onUploadsManagerError({
+                item,
+                error
+            });
+        } else {
+            onError({
+                file,
+                error
+            });
```

Comment:
I meant can you just keep `onError` and `onUpload` but pass in what you'd pass in for the UploadsManager using those callbacks?

e.g.
```
const errorData =  windowView ?
    {
        item,
        error
    } : {
        file,
        error
    };

onError(errorData);
```

Reply:
got it, I tried to avoid that because I thought it's confusing to conditionally set the interfaces of functions. But maybe passing a separate set of props is just as confusing in a slightly different way. We can always split out uploads manager when it's becoming overwhelming... I'll make the change